plugins/fastem_script_runner.py

```diff
@@ -0,0 +1,627 @@
+            command_to_send = self.input_ctrl.GetValue() + os.linesep # Ensure correct OS line ending
+            self.input_ctrl.Clear()
+            try:
+                self.gui_stdout_writer.write(f"[You typed]: {command_to_send}") # Echo to GUI
+                self.process.stdin.write(command_to_send)
+                self.process.stdin.flush()
+            except (OSError, ValueError, BrokenPipeError) as e:
+                self._append_output_safe(f"\n--- Error writing to script input (pipe likely closed): {e} ---\n")
+        event.Skip()
+
+    def on_stop_button_clicked(self, event):
+        if not self.is_script_running or not self.process:
+            return
+
+        if self.stop_button:
+            self.stop_button.Enable(False)  # Disable to prevent multiple clicks
+
+        wx.CallAfter(self._append_output_safe, "\n--- Stop requested. Sending SIGTERM to script... ---\n")
+
+        try:
+            self.process.terminate()
+        except Exception as e:
+            wx.CallAfter(self._append_output_safe, f"\n--- Error sending SIGTERM: {e} ---\n")
+            if self.stop_button:
+                self.stop_button.Enable(True)  # Re-enable if terminate failed
+            return  # Don't start timer if terminate itself failed
+
+        # Start a timer. If SIGTERM doesn't stop it, escalate to SIGKILL.
+        if self.force_kill_timer and self.force_kill_timer.IsRunning():
+            self.force_kill_timer.Stop()
+        self.force_kill_timer = wx.Timer(self, self.TIMER_ID_FORCE_KILL)
+        self.Bind(wx.EVT_TIMER, self.on_force_kill_timeout, id=self.TIMER_ID_FORCE_KILL)
+        self.force_kill_timer.StartOnce(5000) # 5 seconds timeout for SIGKILL
+
+    def on_force_kill_timeout(self, event):
+        self.force_kill_timer = None
+
+        if self.is_script_running and self.process and self.process.poll() is None:
+            wx.CallAfter(self._append_output_safe, "\n--- SIGTERM timed out. Sending SIGKILL to script... ---\n")
+            try:
+                self.process.kill() # Sends SIGKILL on POSIX, TerminateProcess on Windows
+            except Exception as e:
+                wx.CallAfter(self._append_output_safe, f"\n--- Error sending SIGKILL: {e} ---\n")
+
+    def on_close_window(self, event):
+        if self.force_kill_timer and self.force_kill_timer.IsRunning():
+            self.force_kill_timer.Stop()
+            self.force_kill_timer = None
+
+        if self.is_script_running and self.process:
+            wx.CallAfter(self._append_output_safe, "\n--- Window closed. Attempting to terminate script process... ---\n")
+            try:
+                self.process.terminate()
+            except ProcessLookupError:  # Process already gone
+                pass
+            except Exception as e:
+                wx.CallAfter(self._append_output_safe, f"\n--- Error terminating script on close: {e} ---\n")
```

Comment:
Are this `CallAfter()` calls ever going to have an effect? It seems to me that it's show a text on a closed window! Maybe better to use `logging` anyway, in order to have this info in the log files.